src/utils.ts

```diff
@@ -16,28 +18,45 @@ export const getLibraryFolder = () => {
 
 export const getSettings = (): Settings => {
   const settings: any = {};
-  ["PerforceConnectionString", "PerforceTimeout", "PerforceUser"].forEach(key => {
+  ["PerforceConnectionString", "PerforceTimeout", "PerforceUser", "BitbucketOnPremServers"].forEach(key => {
       settings[key] = store.get(key, undefined);
   });
   return settings as Settings;
 };
 
-const overrideGlobalName = (settingKey: string) => `${settingKey}-is-overriding-global`
+const validHttpRegex = /^https?:\/\/(([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})|([\d\w\-_]*(\.[\d\w\-_]+)*))(:([0-9]*))?(\/[A-Za-z0-9]*\/?)*$/;
+
+const areOnPremServersValid = (servers: string[]) => {
+    if (!_.isEmpty(servers)) {
+        if (_.some(servers, server => !validHttpRegex.test(server))) {
+            return false;
+        }
+        return true;
+    }
+};
+
+const overrideGlobalName = (settingKey: string) => `${settingKey}-is-overriding-global`;
 
 export const setSettings = (settings: Settings): Settings => {
-  if (settings.OverrideGlobal) {
-    Object.entries(settings).forEach(([key, val]) => {
-      store.set(key, val)
-      store.set(overrideGlobalName(key), true)
-    });
-  } else {
-    Object.entries(settings).forEach(([key, val]) => {
-      if (store.get(overrideGlobalName(key), false)) {
-        console.info(`skipping setting ${key}:${val} since it was overriden locally`)
-        return
-      }
-      store.set(key, val)
-    });
-  }
-  return  getSettings();
+    if (!areOnPremServersValid(settings?.BitbucketOnPremServers)) {
```

Comment:
move to constraint directive (easily done with https://www.npmjs.com/package/graphql-constraint-directive#pattern)

Reply:
2. no need to always override the servers settings